modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/statechannel/FeeCalculator.scala

```diff
@@ -0,0 +1,95 @@
+package org.tessellation.node.shared.domain.statechannel
+
+import java.math.{MathContext, RoundingMode}
+
+import cats.MonadThrow
+import cats.data.{NonEmptyList, NonEmptySet}
+import cats.syntax.all._
+
+import scala.collection.immutable.SortedMap
+
+import org.tessellation.currency.schema.currency.SnapshotFee
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.schema.balance.Balance
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Interval
+import eu.timepit.refined.types.numeric.{NonNegBigDecimal, NonNegInt, NonNegLong}
+
+case class FeeCalculatorConfig(
+  baseFee: NonNegLong,
+  stakingWeight: NonNegBigDecimal,
+  computationalCost: NonNegLong,
+  proWeight: NonNegBigDecimal
+)
+
+object FeeCalculatorConfig {
+  val noFee: FeeCalculatorConfig =
+    FeeCalculatorConfig(baseFee = 0L, stakingWeight = BigDecimal(0.0), computationalCost = 1L, proWeight = BigDecimal(0.0))
+}
+
+sealed trait FeeCalculator[F[_]] {
+  def calculateRecommendedFee(
+    maybeOrdinal: Option[SnapshotOrdinal],
+    delay: NonNegLong = 0L
+  )(staked: Balance, sizeKb: NonNegInt, tip: NonNegLong = 0L, proScore: FeeCalculator.ProScore = 0.0d): F[SnapshotFee]
+}
+
+object FeeCalculator {
+  def make[F[_]: MonadThrow](configByOrdinal: SortedMap[SnapshotOrdinal, FeeCalculatorConfig]): FeeCalculator[F] =
+    new FeeCalculator[F] {
+      private val allConfigs: NonEmptyList[FeeCalculatorConfig] =
+        NonEmptyList
+          .fromList(configByOrdinal.values.toList)
+          .getOrElse(NonEmptyList.one(FeeCalculatorConfig.noFee))
+
+      private def getConfigByOrdinal(ordinal: SnapshotOrdinal): FeeCalculatorConfig =
+        configByOrdinal
+          .rangeTo(ordinal)
+          .lastOption
+          .map { case (_, config) => config }
+          .getOrElse(FeeCalculatorConfig.noFee)
```

Comment:
Can we move this into the companion object `FeeCalculatorConfig` ? It's something I'll need to use too.